lib/std/private/ospaths2.nim

```diff
@@ -584,15 +584,31 @@ proc searchExtPos*(path: string): int =
     assert searchExtPos("c.nim") == 1
     assert searchExtPos("a/b/c.nim") == 5
     assert searchExtPos("a.b.c.nim") == 5
+    assert searchExtPos(".nim") == -1
+    assert searchExtPos("..nim") == -1
+    assert searchExtPos("a..nim") == 2
 
-  # BUGFIX: do not search until 0! .DS_Store is no file extension!
+  # Unless there is any char that is not `ExtSep` before last `ExtSep` in the file name,
+  # it is not a file extension.
   result = -1
-  for i in countdown(len(path)-1, 1):
+  let stop = when doslikeFileSystem:
+      splitDrive(path).drive.len + 1
```

Comment:
Too slow. Also not necessary as `C:\\` cannot contain a dot.

Reply:
Given `path` can be an UNC path and the server name or the share name in a UNC path can contains '.':
https://learn.microsoft.com/en-us/dotnet/standard/io/file-path-formats
https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-dfsc/149a3039-98ce-491a-9268-2f5ddef08192

Reply:
Maybe it can have another overload with a `allowUNC: bool` ?. 🤔

Reply:
Removed `splitDrive`.